pkg/postgres/pgtest/postgres.go

```diff
@@ -30,27 +30,54 @@ func CreateADatabaseForT(t testing.TB) string {
 	require.NoError(t, err)
 
 	database := strings.ToLower(strings.ReplaceAll(t.Name(), "/", "_") + suffix)
-	// Bootstrap the test database by connecting to the default postgres database and running create
-	sourceWithPostgresDatabase := conn.GetConnectionStringWithDatabaseName("postgres")
-	db, err := sql.Open("postgres", sourceWithPostgresDatabase)
-	require.NoError(t, err)
 
-	_, err = db.Exec("CREATE DATABASE " + database)
+	err = CreateDatabase(t, database)
 	require.NoError(t, err)
-	require.NoError(t, db.Close())
+
 	return database
 }
 
-// DropDatabase drops the named database
-func DropDatabase(t testing.TB, database string) {
+// CreateDatabase - creates a database for testing
+func CreateDatabase(t testing.TB, database string) error {
 	// Bootstrap the test database by connecting to the default postgres database and running create
 	sourceWithPostgresDatabase := conn.GetConnectionStringWithDatabaseName("postgres")
+
 	db, err := sql.Open("postgres", sourceWithPostgresDatabase)
 	require.NoError(t, err)
 
-	_, err = db.Exec("DROP DATABASE " + database)
+	// Checks to see if DB already exists
+	existsStmt := "SELECT EXISTS(SELECT 1 FROM pg_catalog.pg_database WHERE datname = $1)"
+
+	row := db.QueryRow(existsStmt, database)
+	var exists bool
+	if err := row.Scan(&exists); err != nil {
+		exists = false
+	}
+
+	// Only create the test DB if it does not exist
+	if !exists {
+		_, err = db.Exec("CREATE DATABASE " + database)
+		if err != nil {
+			return err
+		}
+	}
 	require.NoError(t, err)
 	require.NoError(t, db.Close())
```

Comment:
ha, well mostly because I copied some of this from test code and some of this from non-test code.  These should be require.NoError to match the rest of the test file.  will resolve.